src/main/java/com/fasterxml/jackson/databind/ser/std/MapSerializer.java

```diff
@@ -1166,23 +1167,57 @@ protected Map<?,?> _orderEntries(Map<?,?> input, JsonGenerator gen,
         if (input instanceof SortedMap<?,?>) {
             return input;
         }
-        // [databind#1411]: TreeMap does not like null key... (although note that
-        //   check above should prevent this code from being called in that case)
-        // [databind#153]: but, apparently, some custom Maps do manage hit this
-        //   problem.
-        if (_hasNullKey(input)) {
-            TreeMap<Object,Object> result = new TreeMap<Object,Object>();
-            for (Map.Entry<?,?> entry : input.entrySet()) {
-                Object key = entry.getKey();
-                if (key == null) {
-                    _writeNullKeyedEntry(gen, provider, entry.getValue());
-                    continue;
+        // or is it empty? then no need to sort either
```

Comment:
I don't think we can ever get `null` values; they are handled separately from regular serialization code flow. So should remove null check.